src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -306,16 +310,31 @@ internal string GetRelatedFileExtensionProperty(string assemblyPath, IEnumerable
             // If no related files found.
             if (relatedFileExtensionList is null || relatedFileExtensionList.Count == 0)
             {
-                _assemblyRelatedExtensions.TryAdd(assemblyPrefix, null);
+                _assemblyRelatedExtensions[assemblyPrefix] = null;
                 return null;
             }
             else
             {
                 relatedFileExtensionList.Sort();
                 string relatedFileExtensionsProperty = string.Join(";", relatedFileExtensionList);
-                _assemblyRelatedExtensions.TryAdd(assemblyPrefix, relatedFileExtensionsProperty);
+                _assemblyRelatedExtensions[assemblyPrefix] = relatedFileExtensionsProperty;
                 return relatedFileExtensionsProperty;
             }
+
+            static bool ReadOnlyMemoryEquals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
```

Comment:
Is it worth making this an extension method that the whole repo can use?

Reply:
Sorry, I had merged before this. 
I think we could eventually. 
You were thinking a build/shared type of method?